cookbook/entities/managing_tasks.rst

```diff
@@ -0,0 +1,254 @@
+        }
+    }
+
+.. _cookbook-entity-template:
+
+The Template
+------------
+
+The template that is responsible to display the form fields should extend the base template
+``OroUIBundle:actions:html.twig`` from the OroUIBundle. This templates defines some basic blocks
+that you can use. This way your own forms will provide the same look and feel as the ones coming
+with the OroPlatform:
+
+.. code-block:: html+jinja
+    :linenos:
+
+    {# src/AppBundle/Resources/views/Task/update.html.twig #}
+
+    {# extend the base template from the OroUIBundle #}
+    {% extends 'OroUIBundle:actions:update.html.twig' %}
+
+    {# reuse the form theme provided with the OroPlatform #}
+    {% form_theme form with 'OroFormBundle:Form:fields.html.twig' %}
+
+    {# make the current task accessible with the task variable #}
+    {% set task = form.vars.value %}
+
+    {# choose the appropriate action depending on whether a task is created or modified #}
+    {# this variable needs to be named formAction as this is what the base template expects #}
+    {% if task.id %}
+        {% set formAction = path('app_task_update', { 'id': task.id }) %}
+    {% else %}
+        {% set formAction = path('app_task_create') %}
+    {% endif %}
+
+    {% block navButtons %}
+        {# the cancelButton() macro creates a button that discards the
+           entered data and leads the user to the linked controller #}
+        {{ UI.cancelButton(path('app_task_index')) }}
+
+        {# the dropdownSaveButton() macro offers a way to let the user select
+           between different options when saving an entity, the selected option
+           will be passed to the controller handling the request as an additonal
+           parameter #}
+        {{ UI.dropdownSaveButton({
+            'html': UI.saveAndCloseButton() ~ UI.saveAndStayButton()
+        }) }}
+    {% endblock navButtons %}
+
+    {% block pageHeader %}
+        {% if task.id %}
+            {% set breadcrumbs = {
+                'entity': task,
+                'indexPath': path('app_task_index'),
+                'indexLabel': 'Tasks',
+                'entityTitle': task.subject
+            } %}
+            {{ parent() }}
+        {% else %}
+            {% set title = 'oro.ui.create_entity'|trans({ '%entityName%': 'Task' }) %}
+            {{ include('OroUIBundle::page_title_block.html.twig', { title: title }) %}
+        {% endif %}
+    {% endblock pageHeader %}
+
+    {% block content_data %}
+        {% set id = 'task-edit' %}
+        {% set dataBlocks = [{
+                'title': 'General'|trans,
+                'class': 'active',
+                'subblocks': [{
+                    'title': '',
+                    'data': [
+                        form_row(form.subject),
+                        form_row(form.description),
+                        form_row(form.dueDate),
+                        form_row(form.priority),
+                    ]
+                }]
+            }]
+        %}
+
```

Comment:
The entity can be extendable. So it can contain custom fields. In this case we should add this custom fields with code

 ```
   {% set additionalData = [] %}
    {% for child in form.children if child.vars.extra_field is defined and child.vars.extra_field %}
        {% set additionalData = additionalData|merge([form_row(child)]) %}
    {% endfor %}
    {% if additionalData is not empty %}
        {% set dataBlocks = dataBlocks|merge([{
            'title': 'Additional'|trans,
            'subblocks': [{
                'title': '',
                'useSpan': false,
                'data' : additionalData
            }]
        }] ) %}
    {% endif %}
```